balancer/rls/picker.go

```diff
@@ -164,29 +197,54 @@ func (p *rlsPicker) delegateToChildPoliciesLocked(dcEntry *cacheEntry, info bala
 			// X-Google-RLS-Data header.
 			res, err := state.Picker.Pick(info)
 			if err != nil {
-				return res, err
+				pr := errToPickResult(err)
+				rf := func() {
+					targetPicksMetric.Record(p.metricsRecorder, 1, p.grpcTarget, p.rlsServerTarget, cpw.target, pr)
+				}
+				if pr == "queue" {
+					// Don't record metrics for queued Picks.
+					rf = func() {}
+				}
```

Comment:
Here and down below, I would pull in the check for queued picks inside the callback:

```
rf := func() {
	if pr == "queue" {
		// Don't record metrics for queued Picks.
		return
	}
	targetPicksMetric.Record(p.metricsRecorder, 1, p.grpcTarget, p.rlsServerTarget, cpw.target, pr)
}
```

The above approach would mean that for completed and failed picks, we would incur the extra cost of running a conditional `if pr == "queue" { ... }`, and branch statements can be expensive with CPU pipelining. I'm not too concerned about it though. I prefer the way the code looks when you pull in it. But if you are @dfawley thinks the cost is not worth the ergonomics, I'm fine with it.

Reply:
Doug said he doesn't mind either, so I switched to this.